student-os/scripts/materials_convert.py

```
@@ -722,6 +865,184 @@ def convert_with_mineru_chunked(
+    if part_repairs:
+        payload["part_repairs"] = part_repairs
+    return payload
+
+
+def convert_with_mineru_agent(source_file: Path, output_path: Path, ctx: ConversionContext) -> dict[str, Any]:
+    file_size = source_file.stat().st_size
+    if file_size > MINERU_AGENT_MAX_FILE_BYTES:
+        raise RuntimeError(
+            f"File exceeds MinerU v1 Agent's {MINERU_AGENT_MAX_FILE_BYTES // (1024 * 1024)}MB size limit; "
+            "provide MINERU_TOKEN to use the v4 precision API."
+        )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)